### PR TITLE
Restore per-folder clear local messages

### LINKED ISSUE #4684: Bring back "clear local messages"
There used to be an option to compact (discard local copies of emails) an imap folder. IIRC it used to be in context menu of folders. It is not available since the UI update. 

This option was quite useful in keeping storage usage in control. Without it my trash folder is getting bigger and bigger.

One option is to add it to left drawer > manage folders > xyz folder > compact folder.

### PR DESCRIPTION
**Pull request by basilgello**
Fixes #4684 

The 'Clear local messages' action is located in toolbar menu of folder setting:

Screenshot_20200428-165122
Screenshot_20200428-165110
Screenshot_20200428-165117

### CODE REVIEW

**Comment by basilgello** [comment]
> Being able to read the text on screenshots is really helpful smile Next time please consider switching the language to English before creating the screenshots for the pull request.

Sorry for that :smile: Will do!

File: app/ui/src/main/res/values-bg/strings.xml [hunk 1]
```diff
@@ -526,6 +526,7 @@ K-9 Mail е мощен, безплатен имейл клиент за Андр
   <string name="folder_settings_folder_notify_mode_first_class">1-ви Клас</string>
   <string name="folder_settings_folder_notify_mode_second_class">2-ри Клас</string>
   <string name="folder_settings_folder_notify_mode_inherited">Същото като push нивото</string>
+  <string name="folder_settings_clear_local_folder_action">Изчисти локалните съобщения</string>
```
[the remark this answers is not in the document]
Comment (basilgello) [reply app/ui/src/main/res/values-bg/strings.xml]: Ok, noted that :)

File: app/ui/src/main/java/com/fsck/k9/ui/managefolders/FolderSettingsFragment.kt [hunk 2]
```diff
@@ -12,15 +19,39 @@ import org.koin.android.ext.android.inject
...
 
+    override fun onCreateOptionsMenu(menu: Menu, inflater: MenuInflater) {
+        super.onCreateOptionsMenu(menu, inflater)
+        inflater.inflate(R.menu.folder_settings_option, menu)
+    }
+
+    override fun onOptionsItemSelected(item: MenuItem): Boolean {
+        return when (item.itemId) {
+            R.id.clear_local_folder -> {
+                onClearFolder()
```
[the remark this answers is not in the document]
Comment (basilgello) [reply app/ui/src/main/java/com/fsck/k9/ui/managefolders/FolderSettingsFragment.kt]: Did it =) It was a nice grep-and-try and exersize ;)

**Comment by basilgello** [comment]
@cketti :)

File: app/ui/src/main/java/com/fsck/k9/ui/managefolders/FolderSettingsFragment.kt [hunk 3]
```diff
@@ -53,10 +80,44 @@ class FolderSettingsFragment : PreferenceFragmentCompat() {
         findPreference<Preference>(PREFERENCE_TOP_CATEGORY)!!.title = folderDisplayName
     }
 
+    private fun handleActionEvents(action: Action) {
+        when (action) {
+            is Action.ClearFolder -> onClearFolder()
+        }
+    }
+
+    private fun onClearFolder() {
```
Comment (cketti) [inline app/ui/src/main/java/com/fsck/k9/ui/managefolders/FolderSettingsFragment.kt:89]: This method should probably be named `showClearFolderConfirmationDialog`.

File: app/ui/src/main/java/com/fsck/k9/ui/managefolders/FolderSettingsViewModel.kt [hunk 4]
```diff
@@ -62,16 +69,36 @@ class FolderSettingsViewModel(
     }
 
     private fun createFolderObject(account: Account, folder: Folder): Folder {
-        val folderType = FolderInfoHolder.getFolderType(account, folder.serverId)
+        folderAccount = account
```
Comment (cketti) [inline app/ui/src/main/java/com/fsck/k9/ui/managefolders/FolderSettingsViewModel.kt:72]: The assignments have nothing to do with creating the `Folder` object. They should be moved outside of this method.

File: app/ui/src/main/java/com/fsck/k9/ui/managefolders/FolderSettingsViewModel.kt [hunk 5]
```diff
@@ -62,16 +69,36 @@ class FolderSettingsViewModel(
...
+        actionLiveData.value = action
+    }
 }
 
 sealed class FolderSettingsResult
 object FolderNotFound : FolderSettingsResult()
 data class FolderSettingsData(val folder: Folder, val dataStore: FolderSettingsDataStore) : FolderSettingsResult()
+
+sealed class Action {
+    object ClearFolder : Action()
```
Comment (cketti) [inline app/ui/src/main/java/com/fsck/k9/ui/managefolders/FolderSettingsViewModel.kt:103]: This should probably be named `ShowClearFolderConfirmationDialog`.

File: app/ui/src/main/java/com/fsck/k9/ui/managefolders/FolderSettingsFragment.kt [hunk 6]
```diff
@@ -53,10 +80,44 @@ class FolderSettingsFragment : PreferenceFragmentCompat() {
...
+        val dialogFragment = ConfirmationDialogFragment.newInstance(
+                DIALOG_CLEAR_FOLDER,
+                getString(R.string.dialog_confirm_delete_title),
```
Comment (cketti) [inline app/ui/src/main/java/com/fsck/k9/ui/managefolders/FolderSettingsFragment.kt:92]: These seem to be the wrong strings. The title is "Confirm deletion", the messasge "Do you want to delete this message?".

**Comment by basilgello** [comment]
@cketti :laughing:

**Comment by cketti** [comment]
k9mail_clear_local_messages

**Comment by basilgello** [comment]
Definitely looks more clear end-user-wise!

**Pull request merged**